napari/_qt/dialogs/_tests/test_qt_plugin_dialog.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+from npe2.manifest.package_metadata import PackageMetadata
+
+from napari._qt.dialogs import qt_plugin_dialog
+
+
+def _hub_or_pypi(conda_forge):
```

Comment:
I know this is not a plugin API or anything, but could you add a docstring? I have no earthly idea what to expect this function to do from the title line, and after reading the implementation I have only a vague earthly idea. :stuck_out_tongue_winking_eye:

Reply:
Absolutely :p

Will do! Short answer, the plugin dialog gets the list of plugins (from pypi or hub) via some generators, so this is creating a mock generator so we do not have to ping neither pypi nor hub when running the tests.

Reply:
Added comments and docstrings!